src/main/java/com/lpnu/shaggybeavers/service/impl/RelicServiceImpl.java

```diff
@@ -17,4 +20,11 @@ public class RelicServiceImpl extends CRUDServiceImpl<Relic, Long> implements Re
     protected JpaRepository<Relic, Long> getRepository () {
         return this.relicRepository;
     }
+
+    @Override
+    public void setImageUrl(URL url, Long relicId) {
+        Relic relic = relicRepository.findById(relicId)
```

Comment:
you can simplifiet it , just set that image url in relic Facade and use service.update but before you neeed to take that relic from bd by id